scripts/checkDeps.ps1

```diff
@@ -40,48 +38,129 @@ if ($? -eq $false || [string]::IsNullOrEmpty($_dockerComposeV)) {
+    $scriptInstalled = (Test-Path ./.conf/.depok) -and (Get-Content ./.conf/.depok | Select-String $script -Quiet)
+
+    if ($scriptInstalled -eq $false) {
+        $_scriptsToInstall.Add($script)
+        Write-Host -ForegroundColor DarkRed "😵 $script dependency installation script not executed before for this project on this machine"
     } else {
-        Write-Host -ForegroundColor DarkGreen "👍 $package installed"
+        Write-Host -ForegroundColor DarkGreen "👍 $script dependency installation script executed before for this project on this machine"
     }
 }
 
-# ask if the user want to install the packages that are not installed
-if ($_packagesToInstall.Count -gt 0) {
+# Check if there are any packages to be installed or scripts to be executed
+if ($_packagesToInstall.Count -eq 0 -and $_scriptsToInstall.Count -eq 0) {
+    Write-Host -ForegroundColor DarkGreen "✅ All packages already installed and installation script executed before for this project on this machine"
+
+    # we need to ran the check deps only if it's not ran yet
+    New-Item -Path .conf/ -Name .depok -ItemType File 2>&1 | out-null
+} else {
+
+    $_packagesInstalledOk = $true
+    $_scriptsInstalledOk = $true
+
+    # ask if the user want to install the packages that are not installed
+    if ($_packagesToInstall.Count -gt 0) {
+
+    $_packagesInstalledOk = $false
+
     $_installConfirm = Read-Host `
-        "Do you want to try to install the dependencies? <y/N>"
-
-    if ($_installConfirm -eq 'y') {
-        # make sure to update the list first
-        sudo apt-get update
-
-        foreach ($item in $_packagesToInstall) {
-            sudo apt-get install -y $item
-            
-            if ($? -eq $false) {
-                Write-Host -ForegroundColor DarkRed "❌ error trying to install package $item"
-                exit 69
+        "Do you want to try to install the missing debian package dependencies? <y/N>"
+
+        if ($_installConfirm -eq 'y') {
+            # make sure to update the list first
+            sudo apt-get update
+
+            $_installedScrips = New-Object Collections.Generic.List[string]
+
+            # Try to install the packages
+            foreach ($item in $_packagesToInstall) {
+                    sudo apt-get install -y $item
+
+                if ($? -eq $false) {
+                    Write-Host -ForegroundColor DarkRed "❌ error trying to install package $item"
+                    exit 69
+                }
+            }
+
+            Write-Host -ForegroundColor DarkGreen "✅ All packages installed successfully"
+
+            $_packagesInstalledOk = $true
+
+        }
+    }
+    # ask if the user want to execute the installation scripts that have not been executed before for this project on this machine
+    if ($_scriptsToInstall.Count -gt 0) {
+
+        $_scriptsInstalledOk = $false
+
+        $_installConfirm = Read-Host `
+            "Do you want to try to install the missing debian package dependencies? <y/N>"
```

Comment:
I think it would be better if we could ask just once, here we can treat it as "dependencies", scripts and packages, and then ask once and install everything.

Reply:
Sounds good.